server/schedule/placement/fit.go

```diff
@@ -110,5 +113,148 @@ func compareRuleFit(a, b *RuleFit) int {
 
 // FitRegion tries to fit peers of a region to the rules.
 func FitRegion(stores core.StoreSetInformer, region *core.RegionInfo, rules []*Rule) *RegionFit {
-	return nil
+	peers := prepareFitPeers(stores, region)
+
+	var regionFit RegionFit
+	if len(rules) == 0 {
+		return &regionFit
+	}
+	for _, rule := range rules {
+		rf := fitRule(peers, rule)
+		regionFit.RuleFits = append(regionFit.RuleFits, rf)
+		// Remove selected.
```

Comment:
does it mean there is only one rule in a range?

Reply:
Not really. There can be multiple rules in a range, i.e. multiple rules for a region.
`Remove selected` here is because a peer can only belong to a Rule.